src/lib.rs

```diff
@@ -890,4 +935,64 @@ mod tests {
             cache.get_or_create(&ids, &path, &fs_attr, false);
         }
     }
+
+    fn do_create_as_ok_test(uid: unistd::Uid, gid: unistd::Gid) {
+        let root = tempdir().unwrap();
+        let file = root.path().join("dir");
+        create_as(&file, uid, gid, |p| fs::create_dir(&p), |p| fs::remove_dir(&p)).unwrap();
+        let fs_attr = fs::symlink_metadata(&file).unwrap();
+        assert_eq!((uid.as_raw(), gid.as_raw()), (fs_attr.uid(), fs_attr.gid()));
+    }
+
+    #[test]
+    fn create_as_self() {
+        do_create_as_ok_test(unistd::Uid::current(), unistd::Gid::current());
+    }
+
+    #[test]
+    fn create_as_other() {
+        if !unistd::Uid::current().is_root() {
+            info!("Test requires root privileges; skipping");
+            return;
+        }
+
+        let config = testutils::Config::get();
+        match config.unprivileged_user {
+            Some(user) => {
+                let uid = unistd::Uid::from_raw(user.uid());
+                let gid = unistd::Gid::from_raw(user.primary_group_id());
+                do_create_as_ok_test(uid, gid);
+            },
+            None => {
+                panic!("UNPRIVILEGED_USER must be set when running as root for this test to run");
```

Comment:
Skip instead?

Reply:
Yes, I'd like to... but how though? I couldn't find a way with the default test runner.

Reply:
Hrm, hrm, maybe just print something and return success? I dunno, this is probably ok.

Reply:
I see. The reason I went with the panic is to make this very obvious and to prevent my CI scripts from missing this case by mistake. I don't expect anyone running these tests as root by hand. Further, `cargo test` swallows the output of successful tests by default, so the printed message wouldn't be visible (tried that before) :-/